jobs/buildhub/inventory_to_records.py

```diff
@@ -443,9 +420,21 @@ def deduplicate_entries(entries):
         batch = []
 
         async for entries in inventory_by_folder(stdin):
-            entries = deduplicate_entries(entries)
-
             for entry in entries:
+                object_key = entry['Key']
+
+                # This is the lowest barrier of entry (no pun intended).
+                # If the entry's 'Key" value doesn't end on any of the
+                # known FILE_EXTENSIONS it will never pass a build URL
+                # later in the loop.
+                if not any(
+                    object_key.endswith(ext) for ext in FILE_EXTENSIONS
+                ):
+                    # Actually, eventually that FILE_EXTENSION check will
+                    # be done again more detailed inside the is_build_url
+                    # function.
+                    # This step was just to weed out some easy ones.
+                    continue
```

Comment:
THESE lines are the most important change. Basically instead of trying to be "too smart" it just does the simplest and fastest of checks. If the file has a decent file extension it gets to participate. Otherwise it's skipped. 

A good thing about this change too is that it removes the need to A) create new list in memory B) create another list from calling `sorted()` and C) not having to create a dict. 

Here's how you can "test" it...

See this file:
https://gist.github.com/peterbe/0b0fb510f3b2b49ab4657db03a83cc9b
If you download it and run it with this horrible script:

```python
import json
from buildhub.utils import key_to_archive_url, is_build_url

PRODUCTS = ('firefox',)
with open('2018-11-13-10-00-51-mozilla-central.entries.log') as f:
    entries = json.load(f)


for thing in entries:
    object_key = thing['Key']
    try:
        # /pub/thunderbird/nightly/...
        product = object_key.split('/')[1]
    except IndexError:
        continue  # e.g. https://archive.mozilla.org/favicon.ico
    if product not in PRODUCTS:
        continue
    url = key_to_archive_url(object_key)
    print(bool(is_build_url(product, url)), url)
```
The output I get is this: https://gist.githubusercontent.com/peterbe/05d803f6830c2292b6dd3d34a2556f6b/raw/d00b2bd5c8549f6f15c1ccddb82c0075e5f003ea/output.log

Reply:
I don't understand do you deduplicate then?

If I remember we had cases where some folder would contain:
- *.exe, *.zip, *.installer.exe 
- only *.installer.exe
- only *.zip

Reply:
You may have been to reproduce the bug you describe by adding some folder entries in this file: https://github.com/mozilla-services/buildhub/blob/master/jobs/tests/data/inventory-simple.csv
For example:
```
pub/firefox/nightly/2018/11/2018-11-13-10-00-51-mozilla-central/firefox-65.0a1.en-US.win32.zip
pub/firefox/nightly/2018/11/2018-11-13-10-00-51-mozilla-central/firefox-65.0a1.en-US.win32.exe
pub/firefox/nightly/2018/11/2018-11-13-10-00-51-mozilla-central/firefox-65.0a1.en-US.win32.installer.exe
pub/firefox/nightly/2018/11/2018-11-13-10-00-51-mozilla-central/firefox-65.0a1.en-US.mac.dmg
pub/firefox/nightly/2018/11/2018-11-13-10-00-51-mozilla-central/firefox-65.0a1.en-US.linux-x86_64.tar.bz2
```
and make it would return:
- `firefox-65.0a1.en-US.win32.exe`
- `firefox-65.0a1.en-US.mac.dmg`
- `firefox-65.0a1.en-US.linux-x86_64.tar.bz2`